execution_engine/src/core/runtime/mod.rs

```diff
@@ -3650,6 +3650,46 @@ where
             }
         }
     }
+
+    fn load_authorization_keys(
+        &mut self,
+        len_ptr: u32,
+        result_size_ptr: u32,
+    ) -> Result<Result<(), ApiError>, Trap> {
+        if !self.can_write_to_host_buffer() {
+            // Exit early if the host buffer is already occupied
+            return Ok(Err(ApiError::HostBufferFull));
+        }
+
+        // A set of keys is converted into a vector so it can be written to a host buffer
+        let authorization_keys =
+            Vec::from_iter(self.context.authorization_keys().clone().into_iter());
+
+        let total_keys = authorization_keys.len() as u32;
+        let total_keys_bytes = total_keys.to_le_bytes();
```

Comment:
Shouldn't we be using `to_bytes()` instead? It will call `to_le_bytes()` on `u32` as well but we would be using a higher-level API that we had defined ourselves.

Reply:
Currently, `bytesrepr` represents u32 as four little-endian bytes, but it may not be the case in the future. The intention is to write u32 as LE bytes under a pointer, rather than 'ABI serialized u32 integer`, which is different (and more expensive). This approach also saves us from calling `bytesrepr::deserialize` on this memory.

Reply:
So what are the rules about this? When do we choose `ToBytes/FromBytes` and when we do not?

Reply:
When you need to pass a single value of a type statically known on the Wasm side like a pointer to u32, then write `to_le_bytes()`, but if you need something more complex like `Vec<AccountHash>`, you should consider `bytesrepr.` Historically we weren't always careful, and we overused `bytesrepr` a lot on the Wasm side, but I hope to address this in 2.0 - for now, I'm trying to stay consistent and pass plain values as LE bytes